### ISSUE TITLE
Java14 Full Records Support Check Validation: MethodCountCheck

### ISSUE THREAD

**Issue by nmancus1** [checkstyle/checkstyle#8518]
Child of #8452 
Check documentation: https://checkstyle.sourceforge.io/config_sizes.html#MethodCountheck

From check documentation:
_Checks the number of methods declared in each type declaration by access modifier or total count._

```
➜  full-record-grammar /usr/lib/jvm/java-14-openjdk/bin/javac --enable-preview --source 14 TestClass.java
Note: TestClass.java uses preview language features.
Note: Recompile with -Xlint:preview for details.
➜  full-record-grammar cat config.xml    
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
      <module name="MethodCount">
          <property name="maxTotal" value="2"/>
      </module>
  </module>
</module>
➜  full-record-grammar cat TestClass.java
class MyTestClass { // violation
    void foo(){}
    void foo2(){}
    void foo3(){}

}

record MyTestRecord() { // should be violation
    void foo(){}
    void foo2(){}
    void foo3(){}
}

//compact ctor
record MyTestRecord2() { // should be ok
    void foo(){}
    void foo2(){}
    public MyTestRecord2{ // make sure no false positive here
    }
}
➜  full-record-grammar java $RUN_LOCALE -jar ~/IdeaProjects/checkstyle/target/checkstyle-8.35-SNAPSHOT-all.jar -c config.xml TestClass.java
Starting audit...
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:1:1: Total number of methods is 3 (max allowed is 2). [MethodCount]
Audit done.
Checkstyle ends with 1 errors.

```
This check needs to be updated to support counting methods within the body of a record definition.

**Comment by esilkensen** [comment]
Fix is merged
